frontend/src/metabase/actions/components/ActionForm/ActionForm.tsx

```diff
@@ -0,0 +1,197 @@
+      formSettings.fields,
+      oldOrder,
+      newOrder,
+    );
+    setFormSettings({
+      ...formSettings,
+      fields: reorderedFields,
+    });
+  };
+
+  const handleChangeFieldSettings = (newFieldSettings: FieldSettings) => {
+    if (!isSettings || !newFieldSettings?.id) {
+      return;
+    }
+
+    setFormSettings({
+      ...formSettings,
+      fields: {
+        ...formSettings.fields,
+        [newFieldSettings.id]: newFieldSettings,
+      },
+    });
+  };
+
+  const handleSubmit = (
+    values: ParametersForActionExecution,
+    actions: FormikHelpers<ParametersForActionExecution>,
+  ) => onSubmit?.(formValidationSchema.cast(values), actions);
+
+  if (isSettings) {
+    return (
+      <FormProvider
+        initialValues={initialValues}
+        validationSchema={formValidationSchema}
+        onSubmit={handleSubmit}
+      >
+        <Form role="form" data-testid="action-form-editor">
+          <DragDropContext onDragEnd={handleDragEnd}>
+            <Droppable droppableId="action-form-droppable">
+              {(provided: DroppableProvided) => (
+                <div {...provided.droppableProps} ref={provided.innerRef}>
+                  {form.fields.map((field, index) => (
+                    <Draggable
+                      key={`draggable-${field.name}`}
+                      draggableId={field.name}
+                      index={index}
+                    >
+                      {(provided: DraggableProvided) => (
+                        <FormFieldContainer
+                          ref={provided.innerRef}
+                          {...provided.draggableProps}
+                          {...provided.dragHandleProps}
+                          isSettings={isSettings}
+                        >
+                          <SettingsContainer>
+                            <Icon name="grabber2" size={14} />
+                          </SettingsContainer>
+
+                          <InputContainer>
+                            <FormFieldWidget
+                              key={field.name}
+                              formField={field}
+                            />
+                          </InputContainer>
+                          <FieldSettingsButtons
+                            fieldSettings={formSettings.fields[field.name]}
+                            onChange={handleChangeFieldSettings}
+                          />
+                        </FormFieldContainer>
+                      )}
+                    </Draggable>
+                  ))}
+                </div>
+              )}
+            </Droppable>
+          </DragDropContext>
+        </Form>
+      </FormProvider>
+    );
+  }
+
+  return (
+    <FormProvider
+      initialValues={initialValues}
+      validationSchema={formValidationSchema}
+      onSubmit={handleSubmit}
+      enableReinitialize
+    >
+      {({ dirty }) => (
+        <Form role="form" data-testid="action-form">
```

Comment:
`disabled` should be also passed on the `Form` or you could submit it with `Enter`